src/gui/MainPane/RoomList.qml

```diff
@@ -73,6 +73,48 @@ HListView {
             keepListCentered = true
     }
 
+    function indexOfRoomId(roomId) {
+        for (let i = 0; i < model.count; i++) {
+            const item = model.get(i)
+            if (item.type === "Room" && item.id === roomId) {
+                return i
+            }
+        }
+        return -1
+
+    }
+
+    // Description can be either room id
+    // or space-separated account id and room id.
+    // If only room id, the first account with this room is used.
+    function showRoomByDescription(description) {
+        const spaceIndex = description.indexOf(" ")
+        if (spaceIndex < 0) {
+            // Description is just room id
+            const roomId = description
+            const roomIndex = indexOfRoomId(roomId)
```

Comment:
You can use `roomList.model.findIndex(roomId)`, no need to make a new function ([code for ModelStore and the model it contains](https://github.com/mirukana/mirage/blob/b62d98ea022e56226f3224acce21f086bfa4709c/src/gui/ModelStore.qml))